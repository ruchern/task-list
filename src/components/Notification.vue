<template>
  <div class="notification" :class="`is-${message.style}`">
    <p class="content" v-html="message.content"></p>
  </div>
</template>

<script>
export default {
  name: 'Notification',
  props: {
    message: Object
  }
}
</script>
