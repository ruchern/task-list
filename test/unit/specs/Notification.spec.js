import { shallowMount } from '@vue/test-utils'
import Notification from '@/components/Notification'

describe('Notification', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Notification, {
      propsData: {
        message: {
          style: 'success'
        }
      }
    })
  })

  it('should render the component', () => {
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.find('div').classes('is-success')).toBeTruthy()
  })
})
