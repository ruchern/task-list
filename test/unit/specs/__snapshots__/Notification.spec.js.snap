// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should render the component 1`] = `
<div
  class="notification is-success"
>
  <p
    class="content"
  />
</div>
`;
